src/dispatchers/playwrightDispatcher.ts

```diff
@@ -40,11 +40,12 @@ export class PlaywrightDispatcher extends Dispatcher<Playwright, channels.Playwr
       selectors: customSelectors || new SelectorsDispatcher(scope, playwright.selectors),
       preLaunchedBrowser,
     }, false);
-    this._portForwardingCallback = portForwardingCallback;
   }
 
   async enablePortForwarding(params: channels.PlaywrightEnablePortForwardingParams): Promise<void> {
-    assert(this._portForwardingCallback, 'Port forwarding is only supported when using connect()');
-    this._portForwardingCallback(params.ports);
+    this._object._setForwardedPorts(params.ports);
+    this._object._portForwardingServer.on('incomingSocksSocket', (socket: SocksInterceptedSocketHandler) => {
```

Comment:
When using the remote server this get's initialized two times, a) by the inprocess and b) by the server. If we have the on(dispatch in the constructor, this results in incorrect behavior. Maybe there is a better way to handle that.

Reply:
Yes currently its like that because the server side Playwright instance get's reused and passed a second time into the PlaywrightDispatcher, see here where it get's reused: https://github.com/microsoft/playwright/blob/0b0512a62492aacccaea7cce475c7000c1772da9/src/browserServerImpl.ts#L48

This results in that we have two times the `incomingSocksSocket` event listener which results in two events getting passed over the protocol. We should probably create a new Playwright event for new Servers created via launchServer.